contrib/encrypt_files_in_dir_to_s3.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python2.7
+# Copyright (C) 2016 UCSC Computational Genomics Lab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Author : Arjun Arkal Rao
+Affiliation : UCSC BME, UCSC Genomics Institute
+File : contrib/encrypt_files_in_dir_to_s3.py
+
+SOURCE: https://github.com/jvivian/one_off_scripts/blob/master/
+        encrypt_files_in_dir_to_s3.py
+ORIGINAL AUTHOR: John Vivian
+
+Move files in a directory, or entire directory structures to S3 with (or without) encryption.
+"""
+from __future__ import print_function
+import argparse
+import base64
+import hashlib
+import os
+import subprocess
+import sys
+import re
+
+
+class InputParameterError(Exception):
+    '''
+    This Error Class will be raised  in the case of a bad parameter provided.
+    '''
+    pass
+
+
+def generate_unique_key(master_key, url):
+    '''
+    This module will take a master key and a url, and then make a new key
+    specific to the url, based off the master.
+    '''
+    with open(master_key, 'r') as keyfile:
+        master_key = keyfile.read()
+    assert len(master_key) == 32, 'Invalid Key! Must be 32 characters. ' \
+        'Key: {}, Length: {}'.format(master_key, len(master_key))
+    new_key = hashlib.sha256(master_key + url).digest()
+    assert len(new_key) == 32, 'New key is invalid and is not ' + \
+        '32 characters: {}'.format(new_key)
+    return new_key
+
+
+def write_to_s3(datum, master_key, bucket, remote_dir):
+    '''
+    This module will take in some datum (a file, or a folder) and write it to
+    S3.  It requires a master key to encrypt the datum with, and a bucket to
+    drop the results into.  If remote dir is set, the datum is dropped into the
+    provided directory.
+    datum - :str: Path to file or folder to write to s3am
+    master_key - :str: Path to master key
+    bucket - :str: Bucket of s3am
+    remote_dir - :str: Path describing pseudo dir to store files on s3
+    '''
+    exit_codes = []
+    s3_url_base = 'https://s3-us-west-2.amazonaws.com/'
```

Comment:
It looks like this script was working with an old version of s3am that was runs as
```
s3am upload [options] SRC_URL BUCKET DST_DIR
```

s3am now runs as 
```
s3am upload [options] SRC_URL DST_DIR
```
where `DST_URL` is of the form `S3://BUCKET/DST_FILENAME` so customizing the region is of no avail.